### PR TITLE
update cardano-deps

### PR DESCRIPTION
**Pull request by NicolasDP**
move all the serde out in a separate utility crate to be used
by the jcli and jormungandr. This crates offer tooling to parse
and generate serde's supported formats. It may be extended to
more in the future

### CODE REVIEW

File: doc/genesis_file.md [hunk 1]
```diff
@@ -52,8 +53,9 @@ There are multiple _parts_ in the genesis file:
 
 | option | format | description |
 |:-------|:-------|:------------|
-| `block0-date` | number | the official start time of the blockchain, in seconds since UNIX EPOCH |
+| `block0_date` | number | the official start time of the blockchain, in seconds since UNIX EPOCH |
 | `discrimination` | string | `production` or `test` |
+| `discrimination` | string | `bft` |
```
Comment (mzabaluev) [inline doc/genesis_file.md]: Surely this should be `block0_consensus`?

File: doc/starting_bft_blockchain.md [hunk 2]
```diff
@@ -24,8 +24,9 @@ Utilising a Round Robin algorithm.
 
 ```yaml
 blockchain_configuration:
-  - [ block0-date, 1550822014 ]
-  - [ discrimination, test ]
+  block0_date: 1550822014
```
Comment (mzabaluev) [inline doc/starting_bft_blockchain.md]: Less weird data format, yay!

File: doc/genesis_file.md (hunk 1 above)
Comment (NicolasDP) [reply doc/genesis_file.md]: thanks for spotting this

File: doc/starting_bft_blockchain.md (hunk 2 above)
Comment (NicolasDP) [reply doc/starting_bft_blockchain.md]: thanks to @CodeSandwich for this new model, it's based on his plan described in rust-cardano#610 . he will update it to better format later.

**Comment by NicolasDP** [comment]
@dkijania , I've made a couple of changes in the integration tests. I noticed that on my machine it was utilising the installed jcli and not the built jcli (which led to misinterpreting the error messages 🤦‍♂️ )

**Pull request merged**